### PR TITLE
Add function to create job via global data store

### PR DESCRIPTION
**Pull request by tianfeng92**
Add function to create job via global data store. Currently both ways are existed and can be controlled by env var ENABLE_DATA_STORE.
Currently even with workable creating job part, the legacy uploading part cannot co-operate with it.

### CODE REVIEW

File: src/reporter.js [hunk 1]
```diff
@@ -15,8 +15,9 @@ const region = process.env.SAUCE_REGION || 'us-west-1'
 
 const api = new SauceLabs({
     user: process.env.SAUCE_USERNAME,
-    key: process.env.SAUCE_ACCESS_KEY,
-    region: region
+    key: "b783ac30-0d4c-4648-a632-b561a6ff7854",
```
Comment (FriggaHel) [inline src/reporter.js:18]: Not sure this should be kept here :)

File: src/reporter.js [hunk 2]
```diff
@@ -94,7 +151,7 @@ module.exports = class TestrunnerReporter {
         /**
          * only upload assets if a session was initiated before
          */
-        if (!sessionId) {
+        if (!sessionId || 0 === sessionId) {
```
Comment (FriggaHel) [inline src/reporter.js:154]: I'm not sure this addition is required

**Comment by alexplischke** [comment]
As discussed earlier offline: Could you add a note in the code that indicates that the new job shell creation is a work-in-progress and not meant for prod?

File: src/reporter.js (hunk 1 above)
Comment (tianfeng92) [reply src/reporter.js:18]: Ah! my debugging info!

File: src/reporter.js (hunk 2 above)
Comment (tianfeng92) [reply src/reporter.js]: removed. Thanks!

**Review by FriggaHel — APPROVED** [review]

**Review by alexplischke — APPROVED** [review]

**Pull request merged**